File: server/internal/models/season.go
package models

import "time"

type Season struct {
	dbModel
	Name          string         `json:"name"`
	Leagues       []League       `json:"leagues"`
	Start         time.Time      `json:"start"`
	End           time.Time      `json:"end"`
	Registrations []Registration `json:"registrations"`
}

File: app/src/models/Season.ts
import type { TimeString } from './types/timeString'
import type { DbModel, Registration, Game, League } from './index'

export interface Season extends DbModel {
  name: string
  start: TimeString
  end: TimeString
  registrations: Array<Registration>
  schedule: Array<Game>
  leagues: Array<League>
}

export type { SeasonId } from './ids'
